bridge/client/app/_components/ktb-stage-overview/ktb-stage-overview.component.ts

```diff
@@ -1,77 +1,103 @@
 import { Project } from '../../_models/project';
+import { Service } from '../../_models/service';
 import { Stage } from '../../_models/stage';
-import { DataService } from '../../_services/data.service';
-import { DtFilterFieldChangeEvent, DtFilterFieldDefaultDataSource } from '@dynatrace/barista-components/filter-field';
 import { ApiService } from '../../_services/api.service';
-import { Service } from '../../_models/service';
-import { DtAutoComplete, DtFilter, DtFilterArray } from '../../_models/dt-filter';
-import { filter, map, switchMap, takeUntil, tap } from 'rxjs/operators';
-import { ActivatedRoute } from '@angular/router';
-import { Subject } from 'rxjs';
-import { DtFilterFieldDefaultDataSourceAutocomplete } from '@dynatrace/barista-components/filter-field/src/filter-field-default-data-source';
+import { DataService } from '../../_services/data.service';
 import { ServiceFilterType } from '../ktb-stage-details/ktb-stage-details.component';
 
 @Component({
   selector: 'ktb-stage-overview',
   templateUrl: './ktb-stage-overview.component.html',
   styleUrls: ['./ktb-stage-overview.component.scss'],
 })
-export class KtbStageOverviewComponent implements OnDestroy, OnInit {
-  public project?: Project;
-  public selectedStage?: Stage;
+export class KtbStageOverviewComponent implements AfterViewInit, OnDestroy {
   public _dataSource = new DtFilterFieldDefaultDataSource();
   public filter: DtFilterArray[] = [];
   public isTriggerSequenceOpen = false;
   private filteredServices: string[] = [];
   private globalFilter: { [projectName: string]: { services: string[] } } = {};
   private unsubscribe$: Subject<void> = new Subject<void>();
 
+  public project$ = this.route.params.pipe(
+    map((params) => params.projectName),
+    filter((projectName) => !!projectName),
+    distinctUntilChanged(),
+    tap(() => {
+      // TODO: kept for the moment
+      this.isTriggerSequenceOpen = this.dataService.isTriggerSequenceOpen;
+      this.dataService.isTriggerSequenceOpen = false;
+    }),
+    switchMap((projectName) => this.dataService.getProject(projectName)),
+    tap((project) => {
+      this.setFilter(project, true);
+    })
+  );
+
+  public readonly selectedStageName$ = this.route.paramMap.pipe(
+    map((params) => params.get('stageName')),
+    withLatestFrom(this.project$),
+    filter(([stageName, project]) => Boolean(stageName && project)),
+    map(([stageName]) => stageName)
+  );
+
+  private readonly paramFilterType$ = this.route.queryParamMap.pipe(map((params) => params.get('filterType')));
+
+  public params$ = combineLatest([this.selectedStageName$, this.paramFilterType$, this.project$]).pipe(
+    tap(([stageName, filterType, project]) => {
+      const stage = project!.getStage(stageName!);
+      if (stage) {
+        this.selectedStageChange.emit({ stage: stage!, filterType: (filterType as ServiceFilterType) ?? undefined });
+      }
+    }),
+    takeUntil(this.unsubscribe$),
+    shareReplay(1)
```

Comment:
In case we don't subscribe in the template, remove the `params$` property and change the subscription in `ngAfterViewInit()`:

```suggestion
  ngAfterViewInit() {
    combineLatest([this.selectedStageName$, this.paramFilterType$, this.project$])
      .pipe(takeUntil(this.unsubscribe$))
      .subscribe(([stageName, filterType, project]) => {
        const stage = project!.getStage(stageName!);
        if (stage) {
          this.selectedStageChange.emit({ stage: stage!, filterType: (filterType as ServiceFilterType) ?? undefined });
        });
    }
```